adobe/spectrum-web-components
# feat(contextual-help): add contextual help pattern

## Description



Adds `@spectrum-web-components/contextual-help`, as per https://spectrum.adobe.com/page/contextual-help/

## Related issue(s)



-  https://github.com/adobe/spectrum-web-components/discussions/4281

## Motivation and context

Contextual help is a pattern used widely across Adobe apps. Even if at its core it is a simple component, the lack of a reusable component around this pattern results in different teams implementing different versions of it.
Differences in implementation differ not only on the visual level, but also on the accessibility and mobile responsiveness level. Current implementations are some of them lacking keyboard navigation support, screen reader support, and not taking into consideration user experience on mobile devices.


## How has this been tested?




    1. Go [here](https://contextual-help--spectrum-web-components.netlify.app/storybook/?path=/story/contextual-help--default)
    2. Open/close the Contextual help component
    3. UI respects the spectrum-css styling
    1. Go [here](https://contextual-help--spectrum-web-components.netlify.app/storybook/?path=/story/contextual-help--default)
    2. Toggle between `info`/`help` variants
    3. Observe the icon is changing from informative to help
    1. Go [here](https://contextual-help--spectrum-web-components.netlify.app/storybook/?path=/story/contextual-help--default)
    2. Change the `placement` prop using the Storybook Controls Addon
    3. Observe the popover is positioned based on the `placement` prop
    1. Go [here](https://contextual-help--spectrum-web-components.netlify.app/storybook/?path=/story/contextual-help--default)
    2. Switch to responsive mode -> choose a mobile view
    3. Observe the Contextual help is opening a dialog on mobile, instead of a popover
    1. Go [here](https://contextual-help--spectrum-web-components.netlify.app/storybook/?path=/story/contextual-help--default)
    2. Check out controlled behaviour by toggling the `open` control

## Screenshots (if appropriate)
![Screenshot 2024-04-22 at 11 28 57](https://github.com/adobe/spectrum-web-components/assets/13311865/c5a0e06f-dca5-4708-ada0-c86ff50a566a)


## Types of changes




## Checklist




## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.
